### ISSUE TITLE
[Feature]: Add a No Escape parameter for Frame5/Dialog5.

### ISSUE THREAD

**Issue by FullBleed** [RPTools/maptool#5037]
### Describe the Problem

The escape key can close frame5 and dialog5 windows without warning.

### The Solution you'd like

Add a noescape parameter to the frame5/dialog5 options.

### Alternatives that you've considered.

Baaaaz has found a js method detailed here: https://discord.com/channels/296230822262865920/1214333938806431804

**Comment by fishface60** [comment]
Could do with something to block a dialog's close button too. I can handle escape with javascript and run code to clean things up when the dialog is cancelled, but if the dialog's closed with the X I can't do anything about it.
I can set noframe=1 so the button doesn't appear, but then it can't be resized.

**Comment by Baaaaaz** [comment]
> Could do with something to block a dialog's close button too. I can handle escape with javascript and run code to clean things up when the dialog is cancelled, but if the dialog's closed with the X I can't do anything about it. I can set noframe=1 so the button doesn't appear, but then it can't be resized.

Yes you can do something about it!  If you take a look at the Lib_GracefulExit.rptok attached on the discord link above, one of the demo macro uses the js `fetch()` with the `keepalive: true` option triggered by a `visibilitychange` event which can be used to run things if a frame5/dialog5 is closed by various means including the X button (well other than PC rebooting or similar! ;)  

The only limitation I've found is that if you reopen a frame (which closes the previous one and triggers the keepalive fetch) there is some latency, e.g. if you wanted to recover form values that have not been submitted.
